nvflare/edge/simulation/devices/pt_cifar10.py

```diff
@@ -0,0 +1,156 @@
+        indices = list(range(len(train_set)))
+        random.shuffle(indices)
+        # select the first subset_size indices
+        indices = indices[: self.subset_size]
+        # create a new train_set from the selected indices
+        train_subset = Subset(train_set, indices)
+        # create a dataloader for the train_subset
+        train_loader = torch.utils.data.DataLoader(train_subset, batch_size=batch_size, shuffle=True, num_workers=2)
+
+        # Network loading
+        net = Cifar10ConvNet()
+        net.load_state_dict(global_model)
+        criterion = torch.nn.CrossEntropyLoss()
+        optimizer = torch.optim.SGD(net.parameters(), lr=0.001, momentum=0.9)
+        net.to(DEVICE)
+
+        # Training loop
+        # Let's do 4 local epochs
+        for epoch in range(4):
+            for i, data in enumerate(train_loader, 0):
+                # get the inputs; data is a list of [inputs, labels]
+                inputs, labels = data[0].to(DEVICE), data[1].to(DEVICE)
+
+                # zero the parameter gradients
+                optimizer.zero_grad()
+
+                # forward + backward + optimize
+                outputs = net(inputs)
+                loss = criterion(outputs, labels)
+                loss.backward()
+                optimizer.step()
+
+                # print the loss every ${subset_size / batch_size}/10 iterations
+                # if i % (self.subset_size / batch_size / 10) == 0:
+                #    print(f"Epoch {epoch}, Step {i}, Loss: {loss.item()}")
+
+        # Calculate the model param diff
+        diff_dict = {}
+        for key, param in net.state_dict().items():
+            numpy_param = param.cpu().numpy() - global_model[key].numpy()
+            # Convert numpy array to list for serialization
+            diff_dict[key] = numpy_param.tolist()
+        return diff_dict
+
+    def process_task(self, task: TaskResponse) -> dict:
+        task_data = task.task_data
+        assert isinstance(task_data, dict)
+        model = from_dict(task_data)
+        if not isinstance(model, DXO):
+            self.logger.error(f"expect model to be DXO but got {type(model)}")
+            raise ValueError("bad model data")
+
+        if model.data_kind != DataKind.WEIGHTS:
+            self.logger.error(f"expect model data kind to be {DataKind.WEIGHTS} but got {model.data_kind}")
+            raise ValueError("bad model data kind")
+
+        global_model = model.data
+        if not isinstance(global_model, dict):
+            self.logger.error(f"expect global_model to be dict but got {type(global_model)}")
+            raise ValueError("bad global model")
+
+        # do the training
+        # convert the global_model to a dict of tensors
+        global_model = {k: torch.tensor(v) for k, v in global_model.items()}
+        diff_dict = self._pytorch_training(global_model)
+
+        # create the result DXO
+        result_dxo = DXO(data_kind=DataKind.WEIGHT_DIFF, data={"dict": diff_dict})
```

Comment:
this key of "dict" needs to match the model updater side.
in future this might needs to be a constant or define some protocol